### PR TITLE
Adds Custom names for all blocks that support it.

### LINKED ISSUE #167: [BUG] Chests, dispensers don't record custom names
**Describe the bug**
Placing, accessing, or breaking a block with a custom name, such as as a chest or dispenser, does not log the name.  If the chest/dispenser is broken and rolled back, the custom name is not set.

**To Reproduce**
Acquire a chest and give it a custom name by using an anvil.  Place, open, then break the chest.  Use /pr l r:10 t:10m and observe that the output does not include the name.  (See the screenshot below, demonstrating this with a dispenser.)  Use /pr rollback r:10 t:10m a:break and observe that the rolled-back chest does not have the name.

**Expected behavior**
The custom name should be logged for place/access/break and should be restored upon a successful rollback.

**Screenshots**
dispenser

**Server (please complete the following information):**
  - Prism Version: 2.1.7 build 144
  - Server Type: Spigot
  - Server Version: git-Spigot-2040c4c-933e909 (MC: 1.15.2)

**Additional context**
Testing was done in Creative mode.

### PR DESCRIPTION
**Pull request by Narimm**
Fixes Block custom names
Fixes deprecation of Door materialdata
Fix NPE on TNT recording.

Closes #167

### CODE REVIEW

File: src/main/java/me/botsko/prism/listeners/PrismEntityEvents.java [hunk 1]
```diff
@@ -819,7 +820,7 @@ public void onEntityExplodeChangeBlock(final EntityExplodeEvent event) {
 
             // don't bother record upper doors.
             if (MaterialTag.DOORS.isTagged(block.getType())) {
-                if (((Door) block.getState().getData()).isTopHalf()) {
+                if (((Door) block.getState().getBlockData()).getHalf() == Bisected.Half.TOP) {
```
Comment (Narimm) [inline src/main/java/me/botsko/prism/listeners/PrismEntityEvents.java:823]: Codacy Issue found: [These nested if statements could be combined](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5490548)

**Comment by Narimm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- src/main/java/me/botsko/prism/actions/BlockAction.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5490548&bid=17992648)

**Comment by Narimm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- src/main/java/me/botsko/prism/actions/BlockAction.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5490548&bid=17992648)

**Comment by Narimm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- src/main/java/me/botsko/prism/actions/BlockAction.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5490548&bid=17992648)

**Comment by Bobcat00** [comment]
I tried testing this PR, and got a bunch of exceptions:

```
2020-05-08 08:12:33 [ERROR] Could not pass event PlayerJoinEvent to Prism v2.1.7-SNAPSHOT-
org.bukkit.event.EventException: null
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:319) ~[spigot-1.15.2.jar:git-Spigot-2040c4c-933e909]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[spigot-1.15.2.jar:git-Spigot-2040c4c-933e909]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:589) [spigot-1.15.2.jar:git-Spigot-2040c4c-933e909]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:576) [spigot-1.15.2.jar:git-Spigot-2040c4c-933e909]
	at net.minecraft.server.v1_15_R1.PlayerList.a(PlayerList.java:176) [spigot-1.15.2.jar:git-Spigot-2040c4c-933e909]
	at net.minecraft.server.v1_15_R1.LoginListener.c(LoginListener.java:154) [spigot-1.15.2.jar:git-Spigot-2040c4c-933e909]
	at net.minecraft.server.v1_15_R1.LoginListener.tick(LoginListener.java:53) [spigot-1.15.2.jar:git-Spigot-2040c4c-933e909]
	at net.minecraft.server.v1_15_R1.NetworkManager.a(NetworkManager.java:220) [spigot-1.15.2.jar:git-Spigot-2040c4c-933e909]
	at net.minecraft.server.v1_15_R1.ServerConnection.c(ServerConnection.java:129) [spigot-1.15.2.jar:git-Spigot-2040c4c-933e909]
	at net.minecraft.server.v1_15_R1.MinecraftServer.b(MinecraftServer.java:1105) [spigot-1.15.2.jar:git-Spigot-2040c4c-933e909]
	at net.minecraft.server.v1_15_R1.DedicatedServer.b(DedicatedServer.java:406) [spigot-1.15.2.jar:git-Spigot-2040c4c-933e909]
	at net.minecraft.server.v1_15_R1.MinecraftServer.a(MinecraftServer.java:984) [spigot-1.15.2.jar:git-Spigot-2040c4c-933e909]
	at net.minecraft.server.v1_15_R1.MinecraftServer.run(MinecraftServer.java:824) [spigot-1.15.2.jar:git-Spigot-2040c4c-933e909]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_251]
Caused by: java.lang.NoClassDefFoundError: com/codahale/metrics/Metric
	at me.botsko.prism.actions.GenericAction.<init>(GenericAction.java:49) ~[?:?]
	at me.botsko.prism.actions.PlayerAction.<init>(PlayerAction.java:3) ~[?:?]
	at me.botsko.prism.actionlibs.ActionFactory.createPlayer(ActionFactory.java:403) ~[?:?]
	at me.botsko.prism.listeners.PrismPlayerEvents.onPlayerJoin(PrismPlayerEvents.java:132) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_251]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_251]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_251]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_251]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:315) ~[spigot-1.15.2.jar:git-Spigot-2040c4c-933e909]
	... 13 more
Caused by: java.lang.ClassNotFoundException: com.codahale.metrics.Metric
	at java.net.URLClassLoader.findClass(Unknown Source) ~[?:1.8.0_251]
	at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:167) ~[spigot-1.15.2.jar:git-Spigot-2040c4c-933e909]
	at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:96) ~[spigot-1.15.2.jar:git-Spigot-2040c4c-933e909]
	at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_251]
	at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_251]
	at me.botsko.prism.actions.GenericAction.<init>(GenericAction.java:49) ~[?:?]
	at me.botsko.prism.actions.PlayerAction.<init>(PlayerAction.java:3) ~[?:?]
	at me.botsko.prism.actionlibs.ActionFactory.createPlayer(ActionFactory.java:403) ~[?:?]
	at me.botsko.prism.listeners.PrismPlayerEvents.onPlayerJoin(PrismPlayerEvents.java:132) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_251]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_251]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_251]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_251]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:315) ~[spigot-1.15.2.jar:git-Spigot-2040c4c-933e909]
	... 13 more
```

**Comment by Narimm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- src/main/java/me/botsko/prism/actions/BlockAction.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5490548&bid=17992648)

**Review by alchemistmatt — DISMISSED** [review]
Looks good.  I'm a bit surprised that you choose to use `(Nameable) state` instead of `(Nameable)state`
but perhaps Java code styles are a bit different than C#

**Comment by Narimm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- src/main/java/me/botsko/prism/actions/BlockAction.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5490548&bid=17992648)

**Review by Bobcat00 — DISMISSED** [review]

**Comment by Narimm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- src/main/java/me/botsko/prism/actions/BlockAction.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5490548&bid=17992648)

**Comment by Narimm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- src/main/java/me/botsko/prism/actions/BlockAction.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5490548&bid=17992648)

**Comment by Narimm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- src/main/java/me/botsko/prism/actions/BlockAction.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5490548&bid=17992648)

**Comment by Narimm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- src/main/java/me/botsko/prism/actions/BlockAction.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5490548&bid=17992648)

**Comment by Narimm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- src/main/java/me/botsko/prism/actions/BlockAction.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5490548&bid=17992648)

**Comment by Bobcat00** [comment]
Custom names are still not shown for a:place

**Pull request merged**